integration_test_suite/integration_tests/src/main/java/io/prometheus/jmx/test/httpserver/general/BlackListAndWhiteListObjectNames_IT.java

```diff
@@ -0,0 +1,210 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prometheus.jmx.test.httpserver.general;
+
+import io.prometheus.jmx.test.DockerImageNameParameters;
+import io.prometheus.jmx.test.HttpClient;
+import io.prometheus.jmx.test.HttpHeader;
+import io.prometheus.jmx.test.Metric;
+import io.prometheus.jmx.test.MetricsParser;
+import io.prometheus.jmx.test.TestUtils;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.antublue.test.engine.api.Parameter;
+import org.antublue.test.engine.api.TestEngine;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BlackListAndWhiteListObjectNames_IT {
+
+    private static Network network;
+
+    private String dockerImageName;
+    private GenericContainer<?> applicationContainer;
+    private GenericContainer<?> exporterContainer;
+    private HttpClient httpClient;
+
+    @TestEngine.ParameterSupplier
+    public static Stream<Parameter> parameters() {
+        return DockerImageNameParameters.parameters();
+    }
+
+    @TestEngine.ParameterSetter
+    public void setParameter(Parameter parameter) {
+        dockerImageName = parameter.value();
+    }
+
+    @TestEngine.BeforeClass
+    public static void beforeClass() {
+        // Shared network
+        network = Network.newNetwork();
+
+        // Get the id to force the network creation
+        network.getId();
+    }
+
+    @TestEngine.BeforeAll
+    public void beforeAll() throws Exception {
+        // Application container
+        applicationContainer = new GenericContainer<>(dockerImageName)
+                .waitingFor(Wait.forLogMessage(".*Running.*", 1))
+                .withClasspathResourceMapping("common", "/temp", BindMode.READ_ONLY)
+                .withClasspathResourceMapping(getClass().getName().replace(".", "/"), "/temp", BindMode.READ_ONLY)
+                .withCommand("/bin/sh application.sh")
+                .withExposedPorts(9999)
+                .withLogConsumer(outputFrame -> System.out.print(outputFrame.getUtf8String()))
+                .withNetwork(network)
+                .withNetworkAliases("application")
+                .withStartupTimeout(Duration.ofMillis(30000))
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .withWorkingDirectory("/temp");
```

Comment:
On my machine (arch linux), I need to increase the number of file descriptors otherwise `firefly:java6` won't start. I guess the default number of file descriptors has changed recently, as this wasn't necessary a couple of months ago.

Anyway, there's a quick fix, just add this:
```
// The firefly/java:6 container needs an increased number of file descriptors, so set the nofile ulimit here.
.withCreateContainerCmdModifier(c -> c.getHostConfig().withUlimits(new Ulimit[]{new Ulimit("nofile", 65536L, 65536L)}))
```

However, when I added this manually to run the tests on my machine, I realized that there's a lot of copy-and-paste. The `applicationContainer` and `exporterContainer` configuration code is copied to 61 files.

It would be good to refactor the tests and move copy-and-paste code to a common helper class to avoid code duplication.

Reply:
I'll add the change.